x/inflation/keeper/inflation.go

```diff
@@ -36,80 +40,97 @@ func (k Keeper) MintCoins(ctx sdk.Context, newCoin sdk.Coin) error {
-	err := k.bankKeeper.SendCoinsFromModuleToModule(ctx, types.ModuleName, k.feeCollectorName, stakingIncentivesCoins)
+	// Allocate staking rewards into fee collector account
+	stakingRewards := sdk.NewCoins(k.GetProportions(ctx, mintedCoin, proportions.StakingRewards))
+	err := k.bankKeeper.SendCoinsFromModuleToModule(
+		ctx,
+		types.ModuleName,
+		k.feeCollectorName,
+		stakingRewards,
+	)
 	if err != nil {
 		return err
 	}
 
-	// TODO replace poolIncentives with usageIncentives
-	// // allocate pool allocation ratio to pool-incentives module account account
-	// poolIncentivesCoins := sdk.NewCoins(k.GetProportions(ctx, mintedCoin, proportions.PoolIncentives))
-	// err = k.bankKeeper.SendCoinsFromModuleToModule(ctx, types.ModuleName, poolincentivestypes.ModuleName, poolIncentivesCoins)
-	// if err != nil {
-	// 	return err
-	// }
-
-	devRewardCoin := k.GetProportions(ctx, mintedCoin, proportions.DeveloperRewards)
-	devRewardCoins := sdk.NewCoins(devRewardCoin)
-	// This is supposed to come from the developer vesting module address, not the mint module address
-	// we over-allocated to the mint module address earlier though, so we burn it right here.
-	err = k.bankKeeper.BurnCoins(ctx, types.ModuleName, devRewardCoins)
+	// Allocate usage incentives to incentives module account
+	usageIncentives := sdk.NewCoins(k.GetProportions(ctx, mintedCoin, proportions.UsageIncentives))
+	err = k.bankKeeper.SendCoinsFromModuleToModule(
+		ctx,
+		types.ModuleName,
+		incentivestypes.ModuleName,
+		usageIncentives,
+	)
 	if err != nil {
 		return err
 	}
-	if len(params.WeightedDeveloperRewardsReceivers) == 0 {
-		// fund community pool when rewards address is empty
-		err = k.distrKeeper.FundCommunityPool(ctx, devRewardCoins, k.accountKeeper.GetModuleAddress(types.DeveloperVestingModuleAcctName))
-		if err != nil {
+
+	// Allocate community pool inflation (remaining module balance) to community
+	// pool address
+	moduleAddr := sdk.AccAddress(types.ModuleAddress.Bytes())
+	communityPool := sdk.NewCoins(k.bankKeeper.GetBalance(ctx, moduleAddr, mintedCoin.Denom))
+	err = k.distrKeeper.FundCommunityPool(
+		ctx,
+		communityPool,
+		k.accountKeeper.GetModuleAddress(types.ModuleName),
+	)
+
+	return err
+}
+
+// AllocateTeamVesting allocates the team vesting proportion from the team
+// vesting supply
+func (k Keeper) AllocateTeamVesting(ctx sdk.Context) error {
+	params := k.GetParams(ctx)
+
+	// Check team vesting account balance
+	if ok := k.bankKeeper.HasSupply(ctx, params.MintDenom); !ok {
+		return sdkerrors.Wrapf(
+			sdkerrors.ErrInsufficientFunds, "team vesting account has no supply",
+		)
+	}
+
+	// Get team vesting provision to allocate from the tharsis account balance. If
+	// tharsis account doesn't have sufficient balance to allocate, only allocate
+	// remaining coins.
+	coin := sdk.NewCoin(params.MintDenom, sdk.NewInt(params.TeamVestingProvision.BigInt().Int64()))
```

Comment:
can you explain what this line is doing? is this converting a `Dec` to an `Int`?

Reply:
@fedekunze yes, do you think the team provision should be set in `int` in the proto instead?

Reply:
change to int